akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1156,15 +1157,20 @@ private[stream] object Collect {
 
       @tailrec private def pushOne(): Unit =
         if (buffer.isEmpty) {
-          if (isClosed(in)) completeStage()
-          else if (!hasBeenPulled(in)) pull(in)
+          if (isClosed(in)) {
+            if (todo == 0) completeStage()
```

Comment:
Is this extra condition needed? Isn't `buffer.isEmpty` and `todo == 0` the same?